### PR TITLE
enhance(controller): modify datasource param and add index for slowsql

### PR DESCRIPTION
**Pull request by goldenxinxing**
### Add index for dataset_read_log
Before:
image
After add index:
image

### Tune datasource param
These exceptions occurs when request increased:
image
image

1. Increase the number of connection pool for hikari(avoid exception of `CannotGetJdbcConnectionException `and `CannotCreateTransactionException`)
2. Increase the time of connection timeout
 
### Log SlowSql
Log like this:

- `2023-10-17 09:02:24.182  INFO [star-whale-controller,2318668389518cb0,2318668389518cb0] 7 --- [io-8082-exec-89] a.s.m.c.SlowSqlLogInterceptor            : Execute SQL?[ select id, role_name, role_code, role_description from user_role_info where id = 1 ] cost[ 254 ms]`
- `2023-10-17 09:03:42.978  INFO [star-whale-controller,629ab1087651b483,c888eb30edefd1c7] 7 --- [io-8082-exec-54] a.s.m.c.SlowSqlLogInterceptor            : Execute SQL?[ SELECT * from dataset_read_log WHERE session_id=551 and (consumer_id is null or consumer_id = '') and status='UNPROCESSED' ORDER BY id LIMIT 1000 ] cost[ 380 ms]`

## TODO

1. Add cache for some stable query.
2. Make Dataset session generation to async.

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        Object parameterObject = boundSql.getParameterObject();
+
+        if (sql == null || sql.length() == 0) {
+            return "";
+        }
+        if (configuration == null) {
+            return "";
+        }
+        TypeHandlerRegistry typeHandlerRegistry = configuration.getTypeHandlerRegistry();
+        sql = sql.replaceAll("[\\s\n ]+", " ");
```
Comment (anda-ren) [inline server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java:118]: `\n` is a subset of `\s`,  isn't `\s+`  sufficient

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java]: fixed

File: server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.springframework.util.ReflectionUtils;
+
+
+@Slf4j
+@Component
+@Intercepts({
+        @Signature(type = StatementHandler.class, method = "query", args = {Statement.class, ResultHandler.class}),
+        @Signature(type = StatementHandler.class, method = "update", args = {Statement.class})
+})
+@ConditionalOnProperty(prefix = "sw.sql.log-slow-sql", name = "enable", havingValue = "true", matchIfMissing = true)
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java:49]: Add support for env variables that users can turn off?

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/configuration/SlowSqlLogInterceptor.java]: done

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**